java/src/jmri/jmrit/display/layoutEditor/LayoutEditorChecks.java

```diff
@@ -0,0 +1,166 @@
+package jmri.jmrit.display.layoutEditor;
+
+import java.awt.event.ActionEvent;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import jmri.util.swing.JMenuUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A collection of tools to check various things on the layout editor panel.
+ */
+public class LayoutEditorChecks {
+
+    private final LayoutEditor layoutEditor;
+    private JMenu checkMenu = new JMenu(Bundle.getMessage("CheckMenuTitle"));
+
+    private JMenu checkUnConnectedTracksMenu = new JMenu(Bundle.getMessage("CheckUnConnectedTracksMenuTitle"));
+    private JMenuItem checkUnConnectedTracksMenuItem = new JMenuItem(Bundle.getMessage("CheckUnConnectedTracksMenuTitle"));
+
+    private JMenu checkUnBlockedTracksMenu = new JMenu(Bundle.getMessage("CheckUnBlockedTracksMenuTitle"));
+    private JMenuItem checkUnBlockedTracksMenuItem = new JMenuItem(Bundle.getMessage("CheckUnBlockedTracksMenuTitle"));
+
+    public LayoutEditorChecks(LayoutEditor editor) {
+        layoutEditor = editor;
+    }
+
+    /**
+     * set the layout editor checks menu (in the tools menu)
+     *
+     * @param toolsMenu where to add our "Check" menu and sub-menus
+     */
+    //Note: initially the sub menu items of the Check menu are JMenuItems with
```

Comment:
What is the value of this switching of items in the menu? It seems to me that allowing users to re-check by clicking the original item would accomplish the same task without introducing inconsistency into the UI.

Reply:
The problem there is that the original menu doesn't have a sub-menu (what would be the point? There's no results in it). Once it's selected it adds the sub-menu with the list of results. Then I realized I needed a way to re-check. I tried it always having the sub-menu with the "Check"/"ReCheck" items in it but this way actually works better. What would be best would be for the checks to run as soon as you mouse over the parent menu and that would populate the sub-menu immediately before it's displayed (Then you wouldn't need the check/recheck + separator). (But I just haven't figured out how to do that yet; I'm guessing there's a menuListener class but I've never used it… ;-)

Reply:
Then I would include a sub menu with "Check X", a separator, and a disabled menu item "No Checks Performed".

> The problem there is that the original menu doesn't have a sub-menu (what would be the point? There's no results in it).

Moving the menu item used to trigger the check (into the submenu) after the first running of the check is arguably being hostile to our users. Telling users explicitly that no checks have been run isn't.

Reply:
Ok, I've implemented the the Check sub-menu's three ways:

 #1) The 1st sub menu "Un-Connected Tracks" has a sub menu that says "Check"; selecting it runs the Un-connected tracks checks and populates that sub-menu. The top sub-menu is renamed to "Re-Check" and there's a divider between it and the results.

#2) the 2nd sub menu just has one menu item that says "Un-Blocked Tracks"; selecting it will run the checks and then add a sub menu with the results. The 1st thing in the results sub-menu will be a "Re-Check" choice followed by a divider between it and any results.

#3) The 3rd menu says "Non-Contiguous Blocks". When this menu is displayed the checks run and immediately populate a sub-menu to this menu with results. There is no "Re-Check" menu because re-opening this menu will trigger the checks to run again and update the sub-menu.

What I don't like about #1 & #2 is that you have to goto the ("[Re-]Check") menu twice: once to (re-)run the check and again to see the results. The #3 solution updates the results menus anytime the "Non-Contiguous Blocks" menu is shown (without a 2nd trip thru the menus to see the results).

Selecting a result from the  "Un-connect Tracks" results sub-menu will select that track in the Layout Editor view.
Selecting a result from the "Un-Blocked Tracks" results sub-menu will open the track editor for that track.
Currently the results from the "Non-Contiguous Blocks" results sub-menu don't do anything (they're explicitly disabled in the code). I'm currently considering ways to display *& fix?)  the non-contiguous blocks...  (Suggestions welcome.)